### PR TITLE
Basic unit testing setup using jest

### PR DESCRIPTION
**Pull request by drenther**
## Subject
Adding basic unit testing structure

## Changes
- added eslint-plugin-jest
- added eslint env jest: true
- ~~added jest as devDep to client~~ using create-react-app jest (changes reverted)
- added jest and supertest as devDep to server
- added dummy test under \__tests\__ in server
- moved mongoose setup to server.js to make app more testable

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -13,6 +13,7 @@
     "redux": "^3.7.2"
   },
   "devDependencies": {
+    "jest": "^22.4.3",
```
Comment (anantoghosh) [inline client/package.json]: create-react-app already installs jest as it's dependency, but it is the older v20 jest.
I'm in favour of using newer jest but I'm not sure if `react-scripts` will pick it up.
Could you run `react-scripts test -v` to check which jest is used?

Comment (drenther) [reply client/package.json]: Yes. I totally forgot about that. No it still will use the old one. If we need the new one, I will edit the npm test script to run the devDep jest (newer version) instead.

Comment (anantoghosh) [reply client/package.json]: I think keep v20. react-scripts 2 already includes jest 22. When it is released we could upgrade if we want. What do you think?

Comment (drenther) [reply client/package.json]: Yes. Less custom config = less confusion. I will just uninstall the jest v22 from client.

File: client/.eslintrc.json [hunk 2]
```diff
@@ -1,10 +1,11 @@
 {
   "env": {
     "es6": true,
-    "browser": true
+    "browser": true,
+    "jest/globals": true
   },
   "extends": ["eslint:recommended", "plugin:react/recommended", "prettier"],
-  "plugins": ["react"],
+  "plugins": ["react", "jest"],
```
Comment (anantoghosh) [inline client/.eslintrc.json]: We will also have to extend `plugin:jest/recommended` to enforce some rules.
Or they have to be specified manually.
If you do extend, keep prettier at last.

Comment (drenther) [reply client/.eslintrc.json]: Done.

**Comment by skyshader** [comment]
Let's add the test script into `pre-commit` hook along with our `lint` check. @drenther and @anantoghosh thoughts on this?

**Comment by drenther** [comment]
@anantoghosh That's right. But I think only the repo owner can setup Travis CI hooks. Right?

**Comment by anantoghosh** [comment]
Yeah, the owner needs to do it.

**Comment by skyshader** [comment]
Yep makes sense. But I don't think that we as moderators have access to set up CI with the repositories. That's the reason I thought it would be better if tested code will only be committed. But yeah, lets not enforce it and mention it in Contributing guide to write and run the tests before submitting the PR. Thoughts? @anantoghosh and @drenther

**Comment by anantoghosh** [comment]
Would it be possible to ask Akshit sir, he just needs to enable the repo in travis. Rest we can do ourselves.
@skyshader Ok

**Comment by anantoghosh** [comment]
This PR is good to merge.
package-lock.json will mess up we will have to handle that separately.

**Pull request merged**